racket/src/cs/rumble/vector.ss

```diff
@@ -568,3 +568,45 @@
    [(size) (make-shared-fxvector size 0)]
    [(size init)
     (register-place-shared (make-fxvector size init))]))
+
+(define/who vector-extend
+  (case-lambda
+   [(v new-size)
+    (vector-extend v new-size 0)]
+   [(v new-size fill)
+    (check who vector? v)
+    (check who exact-nonnegative-integer? new-size)
+    (let ([old-size (vector-length v)])
+      (unless (<= old-size new-size)
+	(raise-arguments-error who
+                               "new length is shorter than existing size"
+                               "new length" new-size 
+                               "existing size" old-size))
```

Comment:
I wonder about "length" versus "size" here. Maybe both should be described as "length"?

Reply:
Ok.